services/galleryService/galleryService.go

```diff
@@ -37,69 +37,77 @@ type Album struct {
 	ImageURL    string
 }
 
-var jsonBytes []byte
-var photos JSONPhotosets
+var albums []map[string]string
 
-func init() {
+const flickrApiURL = "https://api.flickr.com/services/rest/"
 
+func init() {
 	if os.Getenv("FLICKR_API_KEY") == "" {
-		log.Println("Environment variable FLICKR_API_KEY is not assigned")
+		log.Println("Environment variable FLICKR_API_KEY is not assigned.")
 		return
 	}
 
 	if os.Getenv("FLICKR_USER_ID") == "" {
-		log.Println("Environment variable FLICKR_USER_ID is not assigned")
+		log.Println("Environment variable FLICKR_USER_ID is not assigned.")
 		return
 	}
 
-	apiKey := os.Getenv("FLICKR_API_KEY")
-	userID := os.Getenv("FLICKR_USER_ID")
-	hc := http.Client{}
-	apiURL := "https://api.flickr.com/services/rest/"
+	flickrApiKey := os.Getenv("FLICKR_API_KEY")
+	flickrUserID := os.Getenv("FLICKR_USER_ID")
+	httpClient := http.Client{}
 	form := url.Values{}
 	form.Add("method", "flickr.photosets.getList")
 	form.Add("format", "json")
-	form.Add("api_key", apiKey)
-	form.Add("user_id", userID)
+	form.Add("api_key", flickrApiKey)
+	form.Add("user_id", flickrUserID)
 
-	req, err := http.NewRequest("POST", apiURL, strings.NewReader(form.Encode()))
+	req, err := http.NewRequest("POST", flickrApiURL, strings.NewReader(form.Encode()))
 	if err != nil {
 		panic(err)
 	}
 	req.PostForm = form
 	req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
 
-	res, err := hc.Do(req)
+	res, err := httpClient.Do(req)
 	if err != nil {
 		panic(err)
 	}
 	defer res.Body.Close()
 
+	var jsonBytes []byte
 	jsonBytes, err = ioutil.ReadAll(res.Body)
 	if err != nil {
 		panic(err)
 	}
+	// The first 14 characters and the last one are always "jsonFlickrApi(" and ")"
+	// They need to be removed, otherwise the json won't unmarshal
 	jsonBytes = jsonBytes[14 : len(jsonBytes)-1]
+	// The titles and descriptions of the albums returned are stored in fields called
+	// "_content". The underscore makes the unmarshal function ignore them, so they have
+	// to be removed.
```

Comment:
You could change this comment to:

`// Rename field names from "_fieldName" to "fieldName".`